alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -617,24 +605,59 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
 
                 val hits: SearchHits = searchShard(
                     monitorCtx,
-                    concreteIndex,
+                    concreteIndexName,
                     shard,
                     prevSeqNo,
                     maxSeqNo,
                     null,
                     docIds
                 )
+                transformedDocs.addAll(
+                    transformSearchHitsAndReconstructDocs(
+                        hits,
+                        indexName,
+                        concreteIndexName,
+                        monitor.id,
+                        conflictingFields
+                    )
+                )
+            } catch (e: Exception) {
+                logger.error(
+                    "Monitor ${monitor.id} :" +
+                        " Failed to run fetch data from shard [$shard] of index [$concreteIndexName]. Error: ${e.message}",
+                    e
+                )
+            }
+
+            if (transformedDocs.isNotEmpty()) {
+                val matchedQueriesForDocs = getMatchedQueries(
+                    monitorCtx,
+                    transformedDocs.map { it.second },
+                    monitor,
+                    monitorMetadata,
+                    indexName,
+                    concreteIndexName
+                )
+
+                matchedQueriesForDocs.forEach { hit ->
+                    val id = hit.id
+                        .replace("_${indexName}_${monitor.id}", "")
+                        .replace("_${concreteIndexName}_${monitor.id}", "")
```

Comment:
Can we abstract the percolate query execution to a separate function? Makes it modular and easy to understand.

Reply:
this chunk of code is after percolate query execution.

I am renaming the `getMatchedQueries()` to `runPercolateQueryOnTransformedDocs()` for better understanding